Automattic/wp-calypso
# Use redux to control the paid stats upsell modal

Related to https://github.com/Automattic/dotcom-forge/issues/4915

## Proposed Changes

Since we have many entry points to the same Modal, we propose using redux to handle the Modal toggle.
It also solves the issue of when we want to render the Modal within the gated component. See video in the issue, where the modal unmounts along with the component https://github.com/Automattic/wp-calypso/pull/85685#pullrequestreview-1794280163



### What has been implemented so far

In the order of the video below
- Unlock buttons load upsell modal
- DatePicker with Calendar load upsell modal
- The dropdown menu under DatePiacker with Calendar
- Time period buttons load upsell modal (Tabbed) 

https://github.com/Automattic/wp-calypso/assets/6586048/1b6ecee5-9a75-4820-890d-afed1e6f6af1



## Testing Instructions



* Go to /stats with the config flag `?flags=stats/paid-wpcom-v2`, should be automatically turned on for local dev environment
* Go through the items in the video and check that the modals all show correctly

## Pre-merge Checklist
